.github/workflows/emily-cron-ecr-build.yaml

```diff
@@ -0,0 +1,150 @@
+name: Build Docker Image for Emily Cron
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Environment name"
+        required: true
+        default: environment/sbtc-devnet
+        type: choice
+        options:
+          - environment/sbtc-devnet
+          - environment/sbtc-private-mainnet
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: docker-build-emily-cron-${{ github.ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+  attestations: write
+  packages: write
+  issues: read
+  pull-requests: read
+
+env:
+  docker_platforms: "linux/amd64"
+
+jobs:
+  resolve-env:
+    name: Resolve Environment
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ steps.set_env.outputs.environment }}
+      should_run: ${{ steps.set_env.outputs.should_run }}
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Resolve Environment
+        id: set_env
+        run: |
+          ALLOWED_ENVS=("environment/sbtc-devnet" "environment/sbtc-private-mainnet")
+          SHOULD_RUN=true
```

Comment:
nit: we could move this to a reusable workflow or bash script, to avoid the duplication

Reply:
nah just a nit in case we wanted to do something about it now. If we are planning to do something about it in the future it sounds good to me